### PR TITLE
PLT-6722 Convert `fitsInto` tests to golden tests

### PR DESCRIPTION
**Pull request by thealmarty**
This will make it more convenient when the sizes change because we will be able to run `--test-options=--accept` to update the tests instead of doing it manually.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Changelog fragments have been written (if appropriate)
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Targeting master unless this is a cherry-pick backport
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

**Review by michaelpj — COMMENTED** [review]
Looks fine, would just be nice to tweak the test names to avoid weird filenames.

File: plutus-benchmark/nofib/test/Spec.hs [hunk 1]
```diff
@@ -47,7 +47,7 @@ testClausify = testGroup "clausify"
                , testCase "formula4" $ mkClausifyTest Clausify.F4
                , testCase "formula5" $ mkClausifyTest Clausify.F5
                , runTestNested $
-                  Tx.goldenSize "formula1 (size)" (Clausify.mkClausifyCode Clausify.F1)
+                  Tx.goldenSize "formula1-size" (Clausify.mkClausifyCode Clausify.F1)
```
Comment (thealmarty) [inline plutus-benchmark/nofib/test/Spec.hs:50]: the golden files have the suffix `uplc-size`, so I can take out the "size" here if you like. Actually yeah I'll do that.

File: plutus-benchmark/script-contexts/test/checkScriptContext1.uplc-size.golden [hunk 2]
```diff
@@ -0,0 +1 @@
+3432
```
Comment (thealmarty) [inline plutus-benchmark/script-contexts/test/checkScriptContext1.uplc-size.golden:1]: I have verified that `master` has improved the sizes.

File: plutus-benchmark/nofib/test/Spec.hs (hunk 1 above)
Comment (michaelpj) [reply plutus-benchmark/nofib/test/Spec.hs]: Yeah, I was going to suggest that but then the test names are the same as other tests, which does show up in the test output from `tasty`. IDK if it makes anything not work, but it seems maybe bad.

Comment (michaelpj) [reply plutus-benchmark/nofib/test/Spec.hs]: like, if you wanted to rerun just that one test it might be a problem, but that's not such a big deal

Comment (thealmarty) [reply plutus-benchmark/nofib/test/Spec.hs]: hmmm....yeah. I can add the `-size` back in then? or maybe add a `-golden` in?

**Review by michaelpj — APPROVED** [review]
I think the current names are fine but feel free to tweak a bit more if you want.

**Pull request merged**